pkg/endpoint/endpoint.go

```diff
@@ -471,6 +471,8 @@ type Endpoint struct {
 	// is enabled for this endpoint.
 	egressPolicyEnabled bool
 
+	hasBPFProgram chan struct{}
```

Comment:
Replace with an int32

Reply:
This is up to developer preference, so I'm not going to change this. The nice thing about having a channel is that if we want to in the future, we can block on the endpoint not having a BPF program installed yet.